test/unit/testCpu.js

```diff
@@ -56,17 +56,25 @@ exports.cpu_should_not_send_twice_to_same_series = function(finish) {
 
 exports.cpu_should_calculate_percentage_usage = function(finish) {
   var first = true;
+  var mockUsage = [[0, 0], [6, 1000], [0, 0], [6, 2000]];
   function getUsage(cb) {
-    if (first) {
-      first = false;
-      return cb(null, 0);
-    }
-    return cb(null, 60);
+    var usage = mockUsage.shift();
```

Comment:
I'm drawn to this because this `getUsage` function is now mutating state outside of itself. It's fine, since we're only ever using it within this test, but I'm just trying to think of other ways of doing it.

One that seems reasonable is using a generator, but I'm not sure how much better that really is.

Looking around, I can see that it's passed to `metrics.cpu` on line 66 and line 72. I'm starting to get confused now: what will the state of `mockUsage` be in each case? Does each one get its own separate copy of the full thing? Would it be clearer/simpler if we passed a different function as `getUsage` on L72?